### PR TITLE
Add locking to progress tracker and request tracker

### PR DESCRIPTION
**Pull request by ashamis**

### CODE REVIEW

File: src/node/progress_tracker.h [hunk 1]
```diff
@@ -711,13 +630,142 @@ namespace ccf
...
+      LOG_TRACE_FMT(
+        "add_signature node_id:{}, seqno:{}, hashed_nonce:{}",
+        node_id,
+        tx_id.version,
+        hashed_nonce);
+      auto it = certificates.find(tx_id.version);
+      if (it == certificates.end())
+      {
+        // The root is currently not know, so this
+        // signature is saved and and will be verify when the root can be
```
Comment (achamayou) [inline src/node/progress_tracker.h:666]:
```suggestion
        // signature is saved and and will be verified when the root can be
```

File: src/node/progress_tracker.h [hunk 2]
```diff
@@ -711,13 +630,142 @@ namespace ccf
...
+        // retrieved from the primary signature get it from the primary
```
Comment (achamayou) [inline src/node/progress_tracker.h:667]: I'm not quite sure how to parse this?

Comment (ashamis) [reply src/node/progress_tracker.h:667]: oh i really mixed up the tense here :)

File: src/node/progress_tracker.h [hunk 3]
```diff
@@ -711,13 +630,142 @@ namespace ccf
...
+        sig.size());
+      sig_vec.assign(sig.begin(), sig.begin() + signature_size);
+
+      CCF_ASSERT(
+        node_id != id ||
+          std::equal(
+            hashed_nonce.h.begin(),
+            hashed_nonce.h.end(),
+            get_node_hashed_nonce_internal(tx_id).h.begin()),
+        "hashed_nonce does not match my nonce");
```
Comment (achamayou) [inline src/node/progress_tracker.h:719]:
```suggestion
        "hashed_nonce does not match local node nonce");
```

**Review by achamayou — APPROVED** [review]

**Review by jumaffre — APPROVED** [review]

**Pull request merged**